### PR TITLE
azurerm_app_service  - allow v6 IPs for the webapp ip_restriction property

### LINKED ISSUE #6423: Cannot use ipv6 address as ip_restriction in azurerm_app_service
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.19
+ provider.azurerm v2.5.0

### Affected Resource(s)

azurerm_app_service

### Terraform Configuration Files

```hcl
resource "azurerm_app_service" "web" {
  count               = length(var.webapps)
  name                = "webappname"
  location            = "westeurope"
  resource_group_name = azurerm_resource_group.web[0].name
  app_service_plan_id = azurerm_app_service_plan.web[0].id
  https_only          = true

  site_config {

    default_documents = ["index.html"]
    always_on = true

    ip_restriction {
      ip_address  = "x.x.0.0/16" # redacted real ip
    }

    ip_restriction {
      ip_address  = "x:x:x::/44" # redacted real ip
    }
  }

}
```

### Expected Behavior

ipv6 format should be accepted, since move to CIDR notation it doesn't. In 1.44.0 when subnet value was give it worked 

### Actual Behavior

(I've replaced actutal ip values with an x)

```
Error: site_config.0.ip_restriction.1.ip_address must start with IPV4 address and/or slash, number of bits (0-32) as prefix. Example: 127.0.0.1/8. Got "x:x:x::/44".

  on ..\..\..\infra\terraform\global\web\main.tf line 44, in resource "azurerm_app_service" "web":
  44: resource "azurerm_app_service" "web" {
```

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

none

### References

none

### PR DESCRIPTION
**Pull request by andrey-dubnik**
Should resolve #6423

### CODE REVIEW

File: azurerm/internal/services/web/app_service.go [hunk 1]
```diff
@@ -809,7 +809,7 @@ func schemaAppServiceIpRestriction() *schema.Schema {
 				"ip_address": {
 					Type:         schema.TypeString,
 					Optional:     true,
-					ValidateFunc: validate.CIDR,
+					ValidateFunc: validate.CIDRv4n6,
```
Comment (katbyte) [inline azurerm/internal/services/web/app_service.go:812]: I think the `validation.IsCIDR` function should work here? its:
```
// IsCIDR is a SchemaValidateFunc which tests if the provided value is of type string and a valid CIDR
func IsCIDR(i interface{}, k string) (warnings []string, errors []error) {
	v, ok := i.(string)
	if !ok {
		errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
		return warnings, errors
	}

	if _, _, err := net.ParseCIDR(v); err != nil {
		errors = append(errors, fmt.Errorf("expected %q to be a valid IPv4 Value, got %v: %v", k, i, err))
	}

	return warnings, errors
}
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the PR @andrey-dubnik - i think the builtin validation.IsCIDR function should work for this? if so could we use it instead?

thanks!

File: azurerm/internal/services/web/app_service.go (hunk 1 above)
Comment (andrey-dubnik) [reply azurerm/internal/services/web/app_service.go]: As validation package does not have IsCIDR I've added it there and update the schema validation.

**Comment by andrey-dubnik** [comment]
@katbyte any chance for PR to merge or still needs something?

**Review by katbyte — APPROVED** [review]
Hey @andrey-dubnik - just using the plugin SDK IsCIDR instead of writing on into the provider. I've pushed that change and am running the tests to get it merged shortly 🙂

**Pull request merged**